src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,83 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
+    private final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>() {{
+            put(ProductService.class, ProductServiceImpl.class);
+            put(FileReaderService.class, FileReaderServiceImpl.class);
+            put(ProductParser.class, ProductParserImpl.class);
+        }};
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (instances.containsKey(interfaceClazz)) {
+            return instances.get(interfaceClazz);
+        }
+
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + clazz.getName()
+                    + " is not marked with @Component");
+        }
```

Comment:
Ensure that all classes intended for injection are annotated with `@Component`. If any class is missing this annotation, it will cause a `RuntimeException` when attempting to retrieve an instance.